### ISSUE TITLE
unsupported file type: undefined (file: undefined)

### ISSUE THREAD

**Issue by digitalmc** [image-size/image-size#73]
I get the following error when I try to get the image size of select images:

`unsupported file type: undefined (file: undefined)` 

Below are a list of two of the image in question:
- http://jampac.us/wp-content/uploads/2016/04/house_on_fire_benie_sanders.jpg
- http://coffeeshopspanish.com/css-assets/uploads/2016/04/coffee_shop_spanish_new_logo_124x100.png

Most other images work. To note these are both from wordpress sites. The images were saved out of Photoshop (Save for Web & Devices) which means they were properly compressed and then uploaded to Wordpress. 

Any help appreciated!

**Comment by jinzhubaofu** [comment]
+1

**Comment by skyserpent** [comment]
+1

**Comment by anhreg** [comment]
After hours debug, checking the image-resize code and googling around. I found this possibly is the reason:

The server may not support chunk download. Nodejs doesn't recognize this and thinks that the downloading has finished normally, then it triggers the .on('end') event. Thus, the buffer you get is only a few bytes. It doesn't meets the requirement that the jpg TypeMap needs (4096). It will consider this unsupported type. https://github.com/image-size/image-size/blob/master/lib/types/jpg.js

I have tried with your image links. It seems that we have the same issue. Your first image chunk size is 438 bytes. Clearly not enough for image-size to work properly.

Refs: https://github.com/nodejs/node-v0.x-archive/issues/6143 where SteveJobzniak listed out possible causes that the server closes connection in the middle.

I'm looking for workarounds.
